transformer_engine/common/transformer_engine.cpp

```diff
@@ -410,5 +410,4 @@ void nvte_zero_tensor(const NVTETensor tensor, cudaStream_t stream) {
     float zero = 0.0f;
     cudaMemcpyAsync(t.amax.dptr, &zero, sizeof(float), cudaMemcpyHostToDevice, stream);
```

Comment:
The `cudaStreamSynchronize` is required to ensure correct behavior with this `cudaMemcpy`. That said, it seems we can just replace it with a `cudaMemset`: 
```suggestion
    cudaMemsetAsync(t.amax.dptr, 0, sizeof(float), 0stream);
```
@pggPL Do you know if there's a particular reason we're using a `cudaMemcpy` here?

Reply:
Thanks Tim. Updated the PR.

Reply:
Nit: we should remove the `float zero = 0.0f;` as well.

Reply:
I don't know why it was done that way, it is clearly wrong.